src/core/connection.c

```diff
@@ -5626,6 +5656,34 @@ QuicConnProcessUdpUnreachable(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicConnProcessRouteCompletion(
+    _In_ QUIC_CONNECTION* Connection,
+    _In_ const uint8_t* PhysicalAddress,
+    _In_ BOOLEAN Succeeded
+    )
+{
+    if (Succeeded) {
+        QUIC_PATH* Path = &Connection->Paths[0];
+        CxPlatResolveRouteComplete(&Path->Route, PhysicalAddress);
+        QuicSendQueueFlush(&Connection->Send, REASON_ROUTE_COMPLETION);
+    } else {
+        QuicTraceLogConnInfo(
+            Unreachable,
+            Connection,
+            "Route resolution failure");
+        //
+        // Close the connection since the peer is unreachable.
```

Comment:
What if this is just one path that's unreachable?

Reply:
What do you mean? The else path would be reached only if route/arp look-up failed fatally.

Reply:
QUIC supports multiple parallel paths simultaneously, possibly on different interfaces. What if one path succeeds but another fails? We shouldn't kill the connection because of one path.

Reply:
Interesting. I have not groked the parallel paths logic. I wonder how it works with the current code (without this PR) or if it's already broken with the current design (blocking arp in send alloc). Switching interface from xdp to non-xdp is definitely broken.

Reply:
I _think_ multiple XDP interfaces would still work as is. For the multi-path stuff, ping me and we can talk.